src/auth/AuthServiceHandler.cc

```diff
@@ -26,6 +27,8 @@ AuthServiceHandler *get_auth_service_handler(int type, CephContext *cct, KeyServ
     return new CephxServiceHandler(cct, ks);
   case CEPH_AUTH_NONE:
     return new AuthNoneServiceHandler(cct);
+  case CEPH_AUTH_UNKNOWN:
+    return new AuthUnknownServiceHandler(cct);
```

Comment:
i think it's desired to have a null `auth_handler` in `AuthMonitor::prep_auth()`, so we can return `ENOTSUP`, if none of the required auth is supported.